service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
```

Comment:
`self._checkHeartBeat()` sets an interval of 'timeout/2' which could me not enough to make completely sure all workers have enough time to start and send their first heartbeats. Here we add an additional delay (without it I've sometimes seen unnecessary restarts on startup)